src/crate.ts

```diff
@@ -49,6 +50,18 @@ export interface QueryOptions {
 
 const defaultConfig = { sqlEndpoint: 'http://localhost:4200/_sql' }
 
+const retryOptions: Options = {
+  minTimeout: ms('30s'),
+  maxTimeout: ms('1h'),
+  // This translates to retrying for up to 24 hours. It takes 7 retries
+  // to reach the maxTimeout of 1 hour.
+  retries: 30,
+  shouldRetry(error) {
+    // Don't retry if we get this exception
+    return !error.message.includes('DuplicateKeyException')
```

Comment:
Sanity check: are we aware of any other "routine" errors like this where we know we shouldn't retry?

Reply:
The danger here is that we could encounter a situation where it would be better to fail outright, but instead, we'll have to wait 24 hours before the error is surfaced.

On that note: we might want to log a warning each time we retry.

Reply:
We can adjust this over time if needed.